samcli/local/docker/container.py

```diff
@@ -87,7 +87,10 @@ def create(self):
         LOG.info("Mounting %s as %s:ro,delegated inside runtime container", self._host_dir, self._working_dir)
 
         kwargs = {
-            "command": self._cmd,
+            # We run the container with sleep infinity, so the container stays running and we can exec inside it
+            "command": "sleep infinity",
```

Comment:
So we use `sleep infinity` for all containers, even if the `self._persist_container == False`?

Reply:
yes - I'm not 100% sure about this, but the idea is so that there's only 1 code path for invoking a lambda, so we do it the same way both times

Without the persist container option, it simply cleans it up immediately after